src/ess/powder/correction.py

```diff
@@ -260,6 +318,27 @@ def _shallow_copy(da: sc.DataArray) -> sc.DataArray:
     return out
 
 
+class RunNormalization(enum.Enum):
+    """Type of normalization applied to each run."""
+
+    monitor_histogram = enum.auto()
+    monitor_integrated = enum.auto()
+    proton_charge = enum.auto()
+
+
+def insert_run_normalization(
+    workflow: sciline.Pipeline, run_norm: RunNormalization
+) -> None:
+    """Insert providers for a specific normalization into a workflow."""
+    match run_norm:
+        case RunNormalization.monitor_histogram:
+            workflow.insert(normalize_by_monitor_histogram)
+        case RunNormalization.monitor_integrated:
+            workflow.insert(normalize_by_monitor_integrated)
+        case RunNormalization.proton_charge:
+            workflow.insert(normalize_by_proton_charge)
```

Comment:
Maybe I lack some context, but I am not convinced by the design choice here. Why do we need an enum which then decides which providers to insert? Wouldn't it be more natural and a more modular design if we had a small normalization workflow (or 3 actually) which can be passed to (dependency injected into) the main workflow creation?

Reply:
Yes, that would be great. But it requires https://github.com/scipp/sciline/issues/180

Reply:
If we could do this, would you require the users to know where to insert the workflow? I.e.
```python
wf = DreamWorkflow()
norm = MonitorHistogramNormalization()
wf[NormalizedRunData[RunType]] = norm
```
Or do we still need a helper function?
```python
wf = DreamWorkflow()
norm = MonitorHistogramNormalization()
insert_run_normalization(wf, norm)
```

Reply:
Neither:
```python
wf = DreamWorkflow(normalization_workflow=MonitorHistogramNormalization())
```
or something like that.

We should look into scipp/sciline#180 then, before adding workarounds.

Reply:
As upstream changes ran into trouble, I suggest we keep this as it is.